### ISSUE TITLE
Container Options > Command Line... raises critical error

### ISSUE THREAD

**Issue by JJKylee** [staxrip/staxrip#738]
- StaxRip version: 2.6.21 (and 2.6.0 as well)

While inspecting the command line of `MP4Box` to see if #737 can be reproduced,

![StaxRip_2 6 21_MP4Box_command-line_check_critical-bug_20210708](https://user-images.githubusercontent.com/3157518/125013808-5c2e1a80-e021-11eb-8496-7d7fde3b3a5b.png)

I ran into this critical error:

```
Win32Exception (v2.6.21)

The system cannot find the file specified

System.ComponentModel.Win32Exception (0x80004005): The system cannot find the file specified
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at StaxRip.GlobalClass.Execute(String fileName, String arguments) in D:\Dev\staxrip\Source\General\GlobalClass.vb:line 130
   at StaxRip.GlobalClass.ShowCommandLinePreview(String title, String value) in D:\Dev\staxrip\Source\General\GlobalClass.vb:line 935
   at StaxRip.MuxerForm.bnCommandLinePreview_Click(Object sender, EventArgs e) in D:\Dev\staxrip\Source\Forms\MuxerForm.vb:line 1169
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at StaxRip.UI.ButtonEx.OnClick(EventArgs e) in D:\Dev\staxrip\Source\UI\Controls\Controls.vb:line 1950
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

This happens regardless of whether a source file is loaded or not. And also regardless of whether the muxer is `MP4Box` or `mkvmerge`.

**Comment by Dendraspis** [comment]
@JJKylee 
Do you have **Windows Terminal** installed?

**Comment by JJKylee** [comment]
@Dendraspis, yes.

Per your hint, I searched for a resolution and found [this thread](https://stackoverflow.com/questions/63303785/windows-terminals-wt-exe-cannot-be-found-for-non-admin-users-path) on StackOverflow. It works OK now.

Thanks.

**Comment by JJKylee** [comment]
> The question is, if we can close this for now... 🤔

I edited my post. This _empty_ Windows Terminal PowerShell session triggered by StaxRip is _not_ what I expected. Is there a reason why this happens? 🙄❓

**Comment by Dendraspis** [comment]
Of course not.
The problem seems to be, that you're using a non-admin account and StaxRip stores the whole command as an user environment variable. Seems that doesn't work properly.

Have to check and think about some things .....

**Comment by JJKylee** [comment]
Actually, I was expecting something like _Code Preview_ - the `MP4Box` or `mkvmerge` command line displayed in a dialog, which can be copy-and-pasted in a text editor and then can be edited and run in a console.

I think it's better than the current _user environment_ thingy. Is it hard to implement? 🙄

**Comment by Dendraspis** [comment]
I know exactly what you mean, but as for now, the code is executed/shown in Powershell/WT .... which is basically okay, but the environment variable is a bit weird, whereas it should have its purpose. 

I'll find a solution, don't worry.  😁

**Comment by Dendraspis** [comment]
For now, I made the Windows Terminal Preference optional in the settings. 
This should prevent such issues with missing necessary permissions.

**Comment by JJKylee** [comment]
> This should prevent such issues with missing necessary permissions.

Yeah, seems like it, but I'd prefer a copiable command line a la _Preview Code..._. Isn't it better? 🤔

**Comment by Dendraspis** [comment]
It is doable, but for most users not that great, because the Code Preview does not have an auto wrap feature ... so the whole command line is placed in one line. That kills the overview respectively the sense of the preview a bit.

> but I'd prefer a copiable command line
You should be able to select the whole command in the Powershell window and copy it.  🤔

**Comment by JJKylee** [comment]
I've observed a very weird behavior of Windows Terminal regarding `Prefer Windows Terminal over PowerShell if present` option in _Settings_ > _System_. (v2.6.33)

It shows the command line correctly, but _when Windows Terminal is already open with any kind of shell_, a `PowerShell` tab opens **with an empty command line**. (The prompt is set to `%startup_dir%`, though.) Except for this particular case, I can see the full command line of the muxer.

I always keep Windows Terminal open, so I couldn't see the command line of the muxer in the first attempt. That's why I talked about the _Code Preview_ thingy. 😂

Well, I guess it's more like a bug in Windows Terminal. Since it's not critical, let's just ignore it for now. 😅

Thanks for your work. 😉
